ToDoList-frontend/src/app/interceptors/error.interceptor.ts

```diff
@@ -4,16 +4,38 @@ import {
   HttpInterceptor,
   HttpHandler,
   HttpRequest,
+  HttpErrorResponse,
 } from "@angular/common/http";
 import { Observable, throwError } from "rxjs";
 import { catchError } from "rxjs/operators";
+import { ToastrService } from "ngx-toastr";
 
 @Injectable({ providedIn: "root" })
 export class ErrorInterceptor implements HttpInterceptor {
+  constructor(
+    private toastr: ToastrService
+    ) { }
+    
   intercept(
     req: HttpRequest<any>,
     next: HttpHandler
   ): Observable<HttpEvent<any>> {
-    return next.handle(req).pipe(catchError((err) => throwError(err.error)));
+    return next.handle(req).pipe(catchError((error: HttpErrorResponse) => {
+      this.handleErrorResponse(error);
+      return throwError(error);
+    } 
+     ));
+  }
+
+  private handleErrorResponse(error: HttpErrorResponse): void {
+    if (error.status === 404) {
```

Comment:
jesli status bedzie 404, to wyswietla sie 2 toasty: resource no found i request failed. To ficzer? :P